### DISCUSSION TITLE
When will Adonisjs Drive and Ally will be ready?

### DISCUSSION THREAD

**Discussion by faiyaz26** [adonisjs/core#2201] [General]
Hey Guys,
I have developed a website on Adonis V4, want to migrate to V5 but the blocker is the File Driver and Ally module. It seems they are not migrated to V5 yet. Any idea when it will be ready? We just to think our website's development roadmap based on that. Thanks for this great framework :)

**Comment by sjorobekov** [comment]
And Websockets :)

**Comment by McSneaky** [comment]
@sjorobekov WebSocket is there already, but thro providers. The same WS **client** won't come back as it was in V4, since it requires too much maintenance. At least not for now

@faiyaz26 If I recall correctly Ally should land within a month or so (there were some teasers in Twitter) not sure about file drivers, but if you are using S3 or something similar, you can just install S3 client directly. Doing your own file upload is a lot less pain then doing your own social auth 😋

**Reply by faiyaz26** [reply]
Ally one is must for me, file driver is fine for me now. Really waiting for Ally one.

**Reply by carlmathisen** [reply]
> WebSocket is there already, but thro providers. 

@McSneaky Do you know which WS providers exist for V5? I have only had luck finding packages for v4.

**Reply by McSneaky** [reply]
> @McSneaky Do you know which WS providers exist for V5? I have only had luck finding packages for v4.

You can follow this guide: https://preview.adonisjs.com/blog/socketio-with-adonisjs-v5

**Comment by pflirae** [comment]
I am waiting for Drive to be able to move a site from Laravel+Vue to Adonis+Vue, so a date for Drive will be great- to know.

**Reply by thetutlage** [reply]
You can do that already by installing https://github.com/Slynova-Org/flydrive separetely and use it like any other npm dependency

**Reply by pflirae** [reply]
Thanks, will move that direction and start with Adonis+Vue.
